go/tools/builders/env.go

```diff
@@ -158,13 +163,15 @@ func absEnv(envNameList []string, argList []string) error {
 	return nil
 }
 
-func runAndLogCommand(cmd *exec.Cmd, verbose bool) error {
+func runAndLogCommand(cmd *exec.Cmd, stderr *bytes.Buffer, verbose bool) error {
 	if verbose {
 		fmt.Fprintln(os.Stderr, formatCommand(cmd))
 	}
 	cleanup := passLongArgsInResponseFiles(cmd)
 	defer cleanup()
-	if err := cmd.Run(); err != nil {
+	err := cmd.Run()
+	os.Stderr.Write(relativizePaths(stderr.Bytes()))
```

Comment:
In general, use `fmt.Fprint` or `io.Copy` when writing to an `*os.File` or any unknown kind of `io.Writer`. The `Write` may write fewer bytes than its given, and it's important to keep calling it until there's nothing left to write.

Reply:
Good point, done.

Reply:
I was wrong about this: `io.Copy` returns early if a `Write` call doesn't write the entire slice. I was thinking of `Read`, which may return fewer bytes without error. I get them confused, since they're different than the `read` and `write` system calls.

No need to change anything, just wanted to point that out.